src/main/CMV.java

```diff
@@ -108,7 +110,41 @@ private Boolean lic8_calculate() {
         return false;
     }
 
-    private Boolean lic9_calculate() {
+    private boolean lic9_calculate() {
+
+        //prevents array out of bounds (sum of points between and the two points themselves)
+        int points_bound = Parameters.C_PTS + Parameters.D_PTS + 2;
+        double pi = Math.PI;
+        double ep = Parameters.EPSILON;
+
+        if(datapoints.length < 5){
+            return false;
+        }
```

Comment:
Add checks for when:

* `C_PTS < 1`
* `D_PTS < 1`
* `C_PTS + D_PTS > datapoints.length - 3`